dali/pipeline/operators/geometric/bb_flip_test.cc

```diff
@@ -41,205 +49,124 @@ constexpr int kTestDataSize = 10;
  * 1 -> reference data horizontally flipped
  * 2 -> reference data vertically flipped
  */
-using TestData = std::array<std::tuple<Roi, Roi, Roi>, kTestDataSize>;
-
-constexpr TestData wh_rois{
-        {
-                std::tuple<Roi, Roi, Roi>{{.2, .2, .4, .3},
-                                          {.4, .2, .4, .3},
-                                          {.2, .5, .4, .3}},
-                std::tuple<Roi, Roi, Roi>{{.0, .0, .5, .5},
-                                          {.5, .0, .5, .5},
-                                          {.0, .5, .5, .5}},
-                std::tuple<Roi, Roi, Roi>{{.3, .2, .1, .1},
-                                          {.6, .2, .1, .1},
-                                          {.3, .7, .1, .1}},
-                std::tuple<Roi, Roi, Roi>{{.0, .0, .2, .3},
-                                          {.8, .0, .2, .3},
-                                          {.0, .7, .2, .3}},
-                std::tuple<Roi, Roi, Roi>{{.0, .0, .1, .1},
-                                          {.9, .0, .1, .1},
-                                          {.0, .9, .1, .1}},
-                std::tuple<Roi, Roi, Roi>{{.5, .5, .1, .1},
-                                          {.4, .5, .1, .1},
-                                          {.5, .4, .1, .1}},
-                std::tuple<Roi, Roi, Roi>{{.0, .6, .7, .4},
-                                          {.3, .6, .7, .4},
-                                          {.0, .0, .7, .4}},
-                std::tuple<Roi, Roi, Roi>{{.6, .2, .3, .3},
-                                          {.1, .2, .3, .3},
-                                          {.6, .5, .3, .3}},
-                std::tuple<Roi, Roi, Roi>{{.4, .3, .5, .5},
-                                          {.1, .3, .5, .5},
-                                          {.4, .2, .5, .5}},
-                std::tuple<Roi, Roi, Roi>{{.25, .25, .5, .5},
-                                          {.25, .25, .5, .5},
-                                          {.25, .25, .5, .5}},
-        }
-};
+using TestData = std::array<std::array<Roi, 3>, kTestDataSize>;
+
+const TestData rois_wh = {{
```

Comment:
You use a hidden constant defined elsewhere (in the definition of TestData type) and then have to have exactly that number of entries. If you specify fewer entries, the compiler will not complain! This is worse than a magic constant - it's a magic constant defined far away.
If you used a C-style array, the size would be adjusted automatically.
Also, the number of curly braces is mind-boggling - C-style arrays don't need double braces.

FYI: The code below compiles
```
    using A = std::array<int, 5>;
    A a = {{ 1, 2 }};
```

Reply:
Done